server/cli/flags.go

```diff
@@ -20,73 +20,93 @@ package cli
+
+		f.StringVar(&ctx.Attrs, "attrs", ctx.Attrs, "specify an ordered, colon-separated list of node "+
+			"attributes. Attributes are arbitrary strings specifying topography or "+
+			"machine capabilities. Topography might include datacenter designation "+
+			"(e.g. \"us-west-1a\", \"us-west-1b\", \"us-east-1c\"). Machine capabilities "+
+			"might include specialized hardware or number of cores (e.g. \"gpu\", "+
+			"\"x16c\"). "+
+			"The relative geographic proximity of two nodes is inferred from the "+
+			"common prefix of the attributes list, so topographic attributes should be "+
+			"specified first and in the same order for all nodes. "+
+			"For example: --attrs=us-west-1b,gpu.")
+
+		f.DurationVar(&ctx.MaxOffset, "max-offset", ctx.MaxOffset, "specify "+
+			"the maximum clock offset for the cluster. Clock offset is measured on all "+
+			"node-to-node links and if any node notices it has clock offset in excess "+
+			"of --max-offset, it will commit suicide. Setting this value too high may "+
+			"decrease transaction performance in the presence of contention.")
+
+		// Gossip flags.
+
+		f.StringVar(&ctx.GossipBootstrap, "gossip", ctx.GossipBootstrap, "specify a "+
+			"comma-separated list of gossip addresses or resolvers for gossip bootstrap. "+
+			"Each item in the list has an optional type: [type=]<address>. "+
+			"Unspecified type means ip address or dns. Type can also be a load balancer (\"lb\"), "+
+			"a unix socket (\"unix\") or, for single-node systems, \"self\".")
+
+		f.DurationVar(&ctx.GossipInterval, "gossip-interval", ctx.GossipInterval,
+			"approximate interval (time.Duration) for gossiping new information to peers.")
+
+		// KV flags.
+
+		f.BoolVar(&ctx.Linearizable, "linearizable", ctx.Linearizable, "enables linearizable behaviour "+
+			"of operations on this node by making sure that no commit timestamp is reported "+
+			"back to the client until all other node clocks have necessarily passed it.")
+
+		// Engine flags.
+
+		f.Int64Var(&ctx.CacheSize, "cache-size", ctx.CacheSize, "total size in bytes for "+
+			"caches, shared evenly if there are multiple storage devices.")
+
+		f.DurationVar(&ctx.ScanInterval, "scan-interval", ctx.ScanInterval, "specify "+
+			"--scan_interval to adjust the target for the duration of a single scan "+
+			"through a store's ranges. The scan is slowed as necessary to approximately"+
+			"achieve this duration.")
+	}
+
+	var clientCmds []*cobra.Command
+	clientCmds = append(clientCmds, kvCmds...)
+	clientCmds = append(clientCmds, rangeCmds...)
+	clientCmds = append(clientCmds, acctCmds...)
+	clientCmds = append(clientCmds, permCmds...)
+	clientCmds = append(clientCmds, zoneCmds...)
+
+	for _, cmd := range clientCmds {
+		cmd.Flags().StringVar(&ctx.Addr, "addr", ctx.Addr, "the address for connection to the cockroach cluster.")
+	}
+
+	for _, cmd := range append(nodeCmds, clientCmds...) {
```

Comment:
Not sure if the `append` is doing what you want. Seems suspicious to be appending to `nodeCmds` here. I think more idiomatic go would be to do something like:

```
  for _, cmds := range [][]*cobra.Command{nodeCmds, clientCmds} {
    for _, cmd := range cmds {
    }
  }
```